fili-core/src/test/groovy/com/yahoo/bard/webservice/util/TableUtilsSpec.groovy

```diff
@@ -153,4 +154,58 @@ class TableUtilsSpec extends  Specification {
         then:
         constrainedInterval == SimplifiedIntervalList.simplifyIntervals([constrainedInterval1, constrainedInterval2])
     }
+
+    def "logicalTableAvailability returns union of all the intervals for the availability"() {
+        given: "two intervals [2017, 2018] and [2018, 2019]"
+        Interval interval1 = new Interval("2017/2018")
+        Interval interval2 = new Interval("2018/2019")
+
+        AvailabilityTestingUtils.TestAvailability availability1 = new AvailabilityTestingUtils.TestAvailability(
+                [Mock(DataSourceName)] as Set,
+                ["availability": [interval1] as Set]
+        )
+        AvailabilityTestingUtils.TestAvailability availability2 = new AvailabilityTestingUtils.TestAvailability(
+                [Mock(DataSourceName)] as Set,
+                ["availability": [interval2] as Set]
+        )
+
+        PhysicalTableSchema physicalTableSchema = Mock(PhysicalTableSchema)
+        physicalTableSchema.getPhysicalColumnName(_ as String) >> ""
+        physicalTableSchema.getColumns() >> Collections.emptySet()
+
+        SimplifiedIntervalList simplifiedIntervalList1 = Mock(SimplifiedIntervalList)
```

Comment:
Instead of creating a Mock of `SimplifiedIntervalList`, should just create two new `SimplifiedIntervalList`
```
        SimplifiedIntervalList simplifiedIntervalList1 = new SimplifiedIntervalList([interval1] as Set)
        SimplifiedIntervalList simplifiedIntervalList2 = new SimplifiedIntervalList([interval2] as Set)
```